### PR TITLE
Lints for CABF SMIME BRs 7.1.2.3.f - EKUs

### PR DESCRIPTION
**Pull request by robplee**
Aiming for all three of the EKU entries from #712

### CODE REVIEW

**Comment by robplee** [comment]
Merging is blocked until this conversation is resolved: https://github.com/zmap/zlint/pull/744#discussion_r1347031645

File: v3/lints/cabf_smime_br/mailbox_validated_enforce_subject_field_restrictions.go [hunk 1]
```diff
@@ -72,12 +72,13 @@ func NewMailboxValidatedEnforceSubjectFieldRestrictions() lint.LintInterface {
...
 
-// CheckApplies returns true if the provided certificate contains one-or-more of the following SMIME BR policy identifiers:
+// CheckApplies returns true if the provided certificate is a subscriber certificate and contains one-or-more of the following
+// SMIME BR policy identifiers:
 //   - Mailbox Validated Legacy
 //   - Mailbox Validated Multipurpose
 //   - Mailbox Validated Strict
 func (l *mailboxValidatedEnforceSubjectFieldRestrictions) CheckApplies(c *x509.Certificate) bool {
-	return util.IsMailboxValidatedCertificate(c)
+	return util.IsMailboxValidatedCertificate(c) && util.IsSubscriberCert(c)
```
Comment (robplee) [inline v3/lints/cabf_smime_br/mailbox_validated_enforce_subject_field_restrictions.go:81]: To address https://github.com/zmap/zlint/pull/713#issuecomment-1698616156 raised by @mtgag

**Comment by robplee** [comment]
@christopher-henderson, would you be able to give this a look over?  I'm starting to work on some KU SMIME lints and it'd be easier if some of the util changes from this PR were on master.

File: v3/lints/cabf_smime_br/smime_legacy_multipurpose_eku_check.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+// shallHaveCrlDistributionPoints - linter to enforce requirement that SMIME certificates SHALL contain emailProtecton EKU
+type legacyMultipurposeEKUCheck struct {
+}
+
+func init() {
+	lint.RegisterCertificateLint(&lint.CertificateLint{
+		LintMetadata: lint.LintMetadata{
+			Name:          "e_smime_legacy_multipurpose_eku_check",
+			Description:   "Strict/Multipurpose and Legacy: id-kp-emailProtection SHALL be present. Other values MAY be present.  The values id-kp-serverAuth, id-kp-codeSigning, id-kp-timeStamping, and anyExtendedKeyUsage values SHALL NOT be present.",
```
Comment (christopher-henderson) [inline v3/lints/cabf_smime_br/smime_legacy_multipurpose_eku_check.go:31]: In general, it's preferred if there is only one `SHALL` clause in a given lint. This is primarily due to the scenario of encountering both errors at the same time, which typically results in one error shadowing the other. Both errors will eventually be addressed, although the second one will only show up after the first one is resolved.

That's not a blocker for this particular lint, I think, but just a consideration for others.

**Review by christopher-henderson — APPROVED** [review]
@robplee indeed! I was holding off on reviewing until I thought you were sure that the PR was ready for review.

If you were ready for a review much earlier, then I ask that in the future you assign me as a reviewer or reach out when you think it will be ready.

**Pull request merged**